src/ORM/DataExtension.php

```diff
@@ -60,22 +60,52 @@ public function augmentWrite(&$manipulation)
     {
     }
 
+    /**
+     * Optional extension hook called by SilverStripe directly before a database write operation.
+     * Any logic expressed here, will be executed by SilverStripe.
+     *
+     * @return void
+     */
     public function onBeforeWrite()
     {
     }
 
+    /**
+     * Optional extension hook called by SilverStripe directly after a database write operation.
+     * Any logic expressed here, will be executed by SilverStripe.
+     *
+     * @return void
+     */
     public function onAfterWrite()
     {
     }
 
+    /**
+     * Optional extension hook called by SilverStripe directly before deleting a database write operation.
+     * Any logic expressed here, will be executed by SilverStripe.
+     *
+     * @return void
+     */
     public function onBeforeDelete()
     {
     }
 
+    /**
+     * Optional extension hook called by SilverStripe directly after deleting a database write operation.
+     * Any logic expressed here, will be executed by SilverStripe.
+     *
+     * @return void
+     */
     public function onAfterDelete()
     {
     }
 
+    /**
+     * Optional extension hook called by SilverStripe directly during a dev/build operation.
+     * Any logic expressed here, will be executed by SilverStripe.
```

Comment:
Here's the doc block from `DataObject::requireDefaultRecords()`:

```php
    /**
     * Add default records to database. This function is called whenever the
     * database is built, after the database tables have all been created. Overload
     * this to add default records when the database is built, but make sure you
     * call parent::requireDefaultRecords().
     *
     * @uses DataExtension->requireDefaultRecords()
     */
```

Perhaps you could use that and adjust it to communicate that it's run in an extension as opposed to directly on the object?